clar2wasm/src/wasm_utils.rs

```diff
@@ -1285,7 +1285,15 @@ pub fn call_function<'a>(
 
     // Call the function
     func.call(&mut store, &wasm_args, &mut results)
-        .map_err(|_| error_mapping::from_runtime_error_code(instance, &mut store))?;
+        .map_err(|e| {
+            if let Some(vm_error) = e.root_cause().downcast_ref::<Error>() {
+                // This is sound since vm_error points
+                // to a properly initialized value of Error.
+                return unsafe { core::ptr::read(vm_error) };
```

Comment:
Not sure it's really sound. If the error contains a box for example, `ptr::read` will just copy the box reference as a new box, and when both box will be dropped you will have a double-freeing of the value it points to.
https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=21b1ca3b33f92ce9d21fcc2cd2615a47

Same issue with other types that frees in `Drop`, like `Vec`.

Reply:
Also, weird that clippy doesn't trigger https://rust-lang.github.io/rust-clippy/master/index.html#/undocumented_unsafe_blocks

Reply:
To avoid the double free, you could replace the value of the current error by another instance of the enum that won't deallocate.
Here is an example: https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=4056c9303966599eec03cc6770abbb94

But obviously, the best would be to make Error cloneable.

Reply:
> But obviously, the best would be to make Error cloneable.

Making `Error` cloneable appears to require significant refactoring in stacks-core due to cascading clones within the `clarity::vm::errors::Error` struct.

Therefore, using `core::ptr::replace` appears to be the most straightforward solution to take ownership of the Error value. Change for that. And thanks for the hint!